js/memory.js

```diff
@@ -1,9 +1,40 @@
 class MemoryGame {
-  constructor(cards){
+  constructor(cards) {
     this.cards = cards;
-    // add the rest of the class properties here
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
-}
\ No newline at end of file
+  shuffleCards() {
+    let j, x, i;
+
+    if (!this.cards) {
+      return undefined;
+    }
+    for (i = this.cards.length - 1; i > 0; i--) {
+      j = Math.floor(Math.random() * (i + 1));
+      x = this.cards[i];
+      this.cards[i] = this.cards[j];
+      this.cards[j] = x;
+    }
+  }
+
+  checkIfPair(card1, card2) {
+    if (card1 === card2) {
+      this.pairsGuessed += 1;
+      this.pairsClicked += 1;
+      return true;
+    } else {
+      this.pairsClicked += 1;
```

Comment:
Line 25 and 28 contain the same code. This violates the Don't Repeat Yourself principle (DRY). Perhaps delete both and place this code on the first line of the function body.